addons/mail/controllers/mail.py

```diff
@@ -121,22 +121,18 @@ def _redirect_to_record(cls, model, res_id, access_token=None, **kwargs):
         elif not record_action['type'] == 'ir.actions.act_window':
             return cls._redirect_to_messaging()
 
-        url_params = {
-            'model': model,
-            'id': res_id,
-            'active_id': res_id,
-            'action': record_action.get('id'),
-        }
+        url = '/odoo/%s/%s' % (model, res_id)
```

Comment:
```suggestion
        url = '/odoo/action-%s/%s' % (record_action.get('id'), res_id)
```

Reply:
Hello @jpp-odoo 
We will always get `None` from `record_action.get('id')` because record_action will never have an 'id' key.
![image](https://github.com/user-attachments/assets/1f89b0a8-fea8-4dbf-bdda-e511aaf14271)

We are at last getting value for record_action from -
https://github.com/odoo/odoo/blob/469dd943c11c30bbd27342c0dc2d61322a419f70/odoo/addons/base/models/ir_ui_view.py#L2687-L2702
which is not returning 'id' key.

So in my opinion we should take value from `record_action.get('res_id')` for action so that we can use action in our url, what are your suggestions on it?

Reply:
Ok, you can leave your first sugestion:
```py
url = '/odoo/%s/%s' % (model, res_id)
```

Reply:
Hello, this should be f-string (this comment is relevant for probably in many places in this diff).